packages/node-sdk/sm/primitives/evm/rpc/paima-l2.ts

```diff
@@ -37,75 +77,73 @@ function* executePaimaL2Input(input: {
   >;
   primitiveName: string;
   signerAddress: `0x${string}`;
-}) {
-  // TODO This is only for batched messages?
-  if (input.nonce) {
-    const [nonceData] = yield* World.resolve(findNonce, {
-      nonce: input.nonce,
-    });
-    if (nonceData) {
-      log.remote(
-        ComponentNames.PAIMA_SYNC,
-        ["paima-l2"],
-        SeverityNumber.INFO,
-        (log) =>
-          log(
-            `Skipping inputData with duplicate nonce: ${
-              JSON.stringify(input.payload)
-            }`,
-          ),
-      );
-      return;
-    }
-    // guarantee we run this no matter if there is an error or a continue
-    yield* World.resolve(insertNonce, {
-      nonce: input.nonce,
-      block_height: input.paima_block_height,
-    });
-  }
-  // const primitiveName = response.output.syncProtocol.payload.primitiveName;
+}): StateUpdateStream<void> {
+  // For EVM we use the lowercase address.
+  const normalizedSignerAddress = input.signerAddress
```

Comment:
you can manually call lowercase and then cast, but fyi you can also do this through Typebox encode/decode via `TypeboxHelpers.Evm.Address`